cg/apps/demultiplex/sample_sheet/sample_sheet_creator.py

```diff
@@ -91,16 +97,21 @@ def create_sample_sheet_content(self) -> List[List[str]]:
 
     def construct_sample_sheet(self) -> List[List[str]]:
         """Construct the sample sheet."""
-        # Create dummy samples for the indexes that are missing
         if self.run_parameters.requires_dummy_samples:
             self.add_dummy_samples()
+            LOG.info("Created dummy samples for the indexes that are missing")
         else:
-            LOG.info("Skip adding dummy samples since run is not WGS")
+            LOG.info("Skipped adding dummy samples since they are not needed")
         self.remove_unwanted_samples()
-        index.adapt_indexes(
-            samples=self.lims_samples,
-            run_parameters=self.run_parameters,
-        )
+        samples_in_lane: List[FlowCellSample]
+        for lane, samples_in_lane in get_samples_by_lane(self.lims_samples).items():
```

Comment:
I prefer creating a variable above for this call to `get_samples_by_lane(self.lims_samples).items()`, it makes the code easier to read and also makes the `samples_in_lane: List[FlowCellSample]` type hint unnecessary since you declare the list with a specific type.

Reply:
```python
samples: List[FlowCellSample] = get_samples_by_lane(self.lims_samples).items()
for lane, samples_in_lane in samples:
```